test/CumulativeMerkle.test.js

```diff
@@ -102,13 +107,16 @@ describe('Cumulative Merkle Distribution', function () {
             const claimAccounts = proofAccounts.slice((proofAccounts.length / 2) * index, (proofAccounts.length / 2) * (index + 1))
             const claimAmounts = Array.from(claimAccounts).map((claimAccount, _) => ethers.BigNumber.from(dist.claims[claimAccount].amount))
             const claimProofs = Array.from(claimAccounts).map((claimAccount, _) => dist.claims[claimAccount].proof)
-            const claimStructs = Array.from(claimAccounts).map((claimAccount, index) => [claimAccount, claimAmounts[index], claimProofs[index]])
-            const prevBalances = await Promise.all(claimAccounts.map(async (claimAccount, _) => await token.balanceOf(claimAccount)))
-            const expBalances = Array.from(prevBalances).map((prevAmmount, index) =>  parseInt(prevAmmount + claimAmounts[index], 10))
+            const claimBeneficiaries = Array.from(claimAccounts).map((claimAccount, _) => dist.claims[claimAccount].beneficiary)
+            const claimStructs = Array.from(claimAccounts).map((claimAccount, index) => [claimAccount, claimBeneficiaries[index], claimAmounts[index], claimProofs[index]])
+            const prevBalances = await Promise.all(claimBeneficiaries.map(async (beneficiary, _) => await token.balanceOf(beneficiary)))
             await merkleDist.batchClaim(merkleRoot, claimStructs)
 
-            const afterBalancesHex = await Promise.all(claimAccounts.map(async (claimAccount, _) => await token.balanceOf(claimAccount)))
+            const afterBalancesHex = await Promise.all(claimBeneficiaries.map(async (beneficiary, _) => await token.balanceOf(beneficiary)))
             const afterBalances = Array.from(afterBalancesHex).map((afterAmmount, _) => parseInt(afterAmmount["_hex"], 16))
```

Comment:
In my opinion it would be preferable to use BigNumber library instead of Integers for tokens amounts since these numbers easily exceeds the maximum value of Integers.